File: waste_mobile/lib/models/notification.dart
// ignore_for_file: non_constant_identifier_names

class Notification {
  int id;
  String type;
  String title;
  String content;
  int? rid;
  DateTime? read_at;
  DateTime? created_at;

  Notification({
    required this.id,
    required this.type,
    required this.title,
    required this.content,
    this.rid,
    this.read_at,
    this.created_at,
  });

  Map<String, dynamic> toJson() => {
        'id': id,
        'type': type,
        'title': title,
        'content': content,
        'rid': rid,
        'read_at': read_at?.toIso8601String(),
        'created_at': created_at?.toIso8601String(),
      };

  factory Notification.fromJson(Map<String, dynamic> snap) => Notification(
        id: snap['id'],
        type: snap['type'],
        title: snap['title'],
        content: snap['content'],
        rid: snap['rid'],
        read_at:
            snap['read_at'] == null ? null : DateTime.tryParse(snap['read_at']),
        created_at: snap['created_at'] == null
            ? null
            : DateTime.tryParse(snap['created_at']),
      );
}

File: waste_pwa1/app/Models/Notification.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Builder;


use Illuminate\Database\Eloquent\Factories\HasFactory;

/**
 * Class Notification
 * @package App\Models
 * @version August 12, 2025, 8:37 pm +08
 *
 * @property \App\Models\User $user
 * @property integer $user_id Хэнээс
 * @property string $type Төрөл
 * @property string $title Гарчиг
 * @property string $content Агуулга
 * @property integer $rid Холоотой айди
 * @property string $read_at Уншсан
 */
class Notification extends Model
{

    use HasFactory;

    use HasFilter;

    public $table = 'notifications';

    const CREATED_AT = 'created_at';
    const UPDATED_AT = 'updated_at';




    public $fillable = [
        'user_id',
        'type',
        'title',
        'content',
        'rid',
        'read_at'
    ];

    /**
     * The attributes that should be casted to native types.
     *
     * @var array
     */
    protected $casts = [
        'id' => 'integer',
        'user_id' => 'integer',
        'type' => 'string',
        'title' => 'string',
        'content' => 'string',
        'rid' => 'integer',
        'read_at' => 'string'
    ];

    /**
     * Validation rules
     *
     * @var array
     */
    public static $rules = [
        'user_id' => 'required',
        'type' => 'required|string|max:100',
        'title' => 'required|string|max:500',
        'content' => 'nullable|string',
        'rid' => 'nullable',
        'read_at' => 'nullable|string',
        'created_at' => 'nullable|string',
        'updated_at' => 'nullable|string'
    ];

    /**
     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
     **/
    public function user()
    {
        return ($this->type=='1')? $this->belongsTo(\App\Models\User::class, 'user_id'): $this->belongsTo(\App\Models\Customer::class, 'user_id');
    }

    /**
     * @var array
     */
    public static $searchIn = [
        'user_id',
        'type',
        'title',
        'content',
        'rid',
        'read_at'
    ];

    /**
     * Filter Model
     *
     * @return array
     */
    public function getSearchIn()
    {
        return Notification::$searchIn;
    }
}
